torch/_refs/__init__.py

```diff
@@ -1171,15 +1171,49 @@ def clamp(
 ) -> TensorLikeType:
     a, min, max = _maybe_broadcast(a, min, max)
 
-    if min is not None and max is not None:
-        return minimum(maximum(a, min), max)
+    # NOTE: grad behavior with implementation `where` is not consistent on `nan`
+    if min is None and max is None:
+        msg = "clamp called but both min and max are none!"
+        raise ValueError(msg)
     if min is not None:
-        return maximum(a, min)
+        a_isnan = isnan(a)
+        condition = bitwise_or(ge(a, min), a_isnan)
```

Comment:
Highlight this section for `nan` propagation. Tagging @ngimel

Reply:
Yeah looks correct